pydy/viz/scene.py

```diff
@@ -48,10 +53,69 @@
         except ImportError:
             from IPython.html import widgets
         from IPython.display import display, Javascript
+        from IPython.utils.traitlets import CFloat, List
 except ImportError:
     IPython = None
 
 
+def quaternion_from_rotation(m):
+    """
+    m is a 3 by 3 matrix, as a list of rows.  The columns of this matrix are
+    the vectors x, y, and z
+    """
+    x = m[0:3]
+    y = m[3:6]
+    z = m[6:9]
+    trace = x[0]+y[1]+z[2]
+    if (trace>0):
+        s = 0.5/sqrt(trace+1)
+        quaternion = [(y[2]-z[1])*s, (z[0]-x[2])*s, (x[1]-y[0])*s, 0.25/s]
+    elif (x[0]>y[1] and x[0]>z[2]):
+        s = 2.0*sqrt(1.0+x[0]-y[1]-z[2])
+        quaternion = [0.25*s, (y[0]+x[1])/s, (z[0]+x[2])/s, (y[2]-z[1])/s]
+    elif (y[1]>z[2]):
+        s = 2.0*sqrt(1.0+y[1]-x[0]-z[2])
+        quaternion = [(y[0]+x[1])/s, 0.25*s, (z[1]+y[2])/s, (z[0]-x[2])/s]
+    else:
+        s = 2.0*sqrt(1.0+z[2]-x[0]-y[1])
+        quaternion = [(z[0]+x[2])/s, (z[1]+y[2])/s, 0.25*s, (x[1]-y[0])/s]
+    return quaternion
+
+
+def _create_pythreejs_geometery(shape):
+    from . import shapes
+    if type(shape) == shapes.Cube:
+        geo = p3js.BoxGeometry(length=shape.length)
+    elif type(shape) == shapes.Cylinder:
+        geo = p3js.CylinderGeometry(radiusTop=shape.radius,
+                                    radiusBottom=shape.radius,
+                                    height=shape.length)
+    elif type(shape) == shapes.Cone:
+        geo = p3js.CylinderGeometry(radiusTop=0,
+                                    radiusBottom=shape.radius,
+                                    height=shape.length)
+    elif type(shape) == shapes.Sphere:
+        geo = p3js.SphereGeometry(radius=shape.radius)
+    elif type(shape) == shapes.Circle:
+        geo = p3js.CircleGeometry(radius=shape.radius, segments=32)
+    elif type(shape) == shapes.Plane:
+        geo = p3js.PlaneGeometry(width=shape.width, length=shape.length)
+    elif type(shape) == shapes.Tetrahedron:
+        geo = p3js.TetrahedronGeometry(radius=shape.radius)
+    elif type(shape) == shapes.Octahedron:
+        geo = p3js.OctahedronGeometry(radius=shape.radius)
+    elif type(shape) == shapes.Icosahedron:
+        geo = p3js.IcosahedronGeometry(radius=shape.radius)
+    elif type(shape) == shapes.Torus:
+        geo = p3js.TorusGeometry(radius=shape.radius, tube=shape.tube_radius)
+    elif type(shape) == shapes.TorusKnot:
+        geo = p3js.TorusKnotGeometry(radius=shape.radius,
+                                     tube=shape.tube_radius)
+    else:
+        raise TypeError('No pythreejs geometry for {}'.format(type(shape)))
+    return geo
```

Comment:
This needs to be removed since this functionality is now baked into `shapes.py`.